assembl/static2/js/app/pages/brightMirrorFiction.jsx

```diff
@@ -1,85 +1,156 @@
-  };
-
-  return (
-    <Fragment>
-      <Grid fluid className="bright-mirror-fiction background-fiction-default">
-        <Row>
-          <Col xs={12}>
-            <article>
-              <FictionHeader {...fictionHeaderProps} />
-              <FictionBody {...fictionBodyProps} />
-            </article>
-          </Col>
-        </Row>
-      </Grid>
-    </Fragment>
-  );
+export type BrightMirrorFictionProps = {
+  /** Fiction data information fetched from GraphQL */
+  data: BrightMirrorFictionData,
+  /** URL slug */
+  slug: string,
+  /** Fiction phase */
+  phase: string,
+  /** Fiction theme identifier */
+  themeId: string,
+  /** Fiction identifier */
+  fictionId: string,
+  /** Fiction locale */
+  contentLocale: string
+};
+type BrightMirrorFictionState = {
+  /** Fiction title */
+  title: string,
+  /** Fiction content */
+  content: string
 };
+const EMPTY_STRING = '';
+export class BrightMirrorFiction extends Component<BrightMirrorFictionProps, BrightMirrorFictionState> {
+  constructor(props: BrightMirrorFictionProps) {
+    super(props);
+    this.state = {
+      title: props.data.fiction.subject ? props.data.fiction.subject : EMPTY_STRING,
+      content: props.data.fiction.body ? props.data.fiction.body : EMPTY_STRING
+    };
+  }
 
+  render() {
+    const { data, slug, phase, themeId, fictionId, contentLocale } = this.props;
+    const { title, content } = this.state;
+    // Handle fetching error
+    if (data.error) {
+      displayAlert('danger', I18n.t('error.loading'));
+      return null;
+    }
+    // Define variables
+    const { fiction } = data;
+    const getDisplayName = () => (fiction.creator && fiction.creator.displayName ? fiction.creator.displayName : EMPTY_STRING);
+    const displayName = fiction.creator && fiction.creator.isDeleted ? I18n.t('deletedUser') : getDisplayName();
+    // Define user permission
+    const USER_ID_NOT_FOUND = -9999;
+    const userId = fiction.creator ? fiction.creator.userId : USER_ID_NOT_FOUND;
+    const userCanDelete =
+      (getConnectedUserId() === String(userId) && connectedUserCan(Permissions.DELETE_MY_POST)) ||
+      connectedUserCan(Permissions.DELETE_POST);
+    const userCanEdit = getConnectedUserId() === String(userId) && connectedUserCan(Permissions.EDIT_MY_POST);
+    // Define callback functions
+    const deleteFictionCallback = () => {
+      // Route to fiction list page
+      const fictionListParams = { slug: slug, phase: phase, themeId: themeId };
+      const fictionListURL = get('idea', fictionListParams);
+      // Set a callback state in order to display a delete fiction confirmation message
+      browserHistory.push({
+        pathname: fictionListURL,
+        state: { callback: FICTION_DELETE_CALLBACK }
+      });
+    };
+    const modifyFictionCallback = (subject, body) => {
+      this.setState({ title: subject, content: body });
+    };
+    // Define components props
+    const circleAvatarProps: CircleAvatarProps = {
```

Comment:
not really since Flow don't really control when you add extra props in a component: that way, we are sure those are the props required by the component.

Reply:
ok, good to know!